lib/topological_inventory/persister/workflow.rb

```diff
@@ -216,6 +231,43 @@ def requeue_sweeping!
       def sweep_retry_count_limit
         100
       end
+
+      def link_data_to_refresh_state_part
+        refresh_state_part = ::RefreshStatePart.where(:uuid => refresh_state_part_uuid).first
+        if refresh_state_part.present?
+          ics = inventory_collections.select { |ic| ic.data.present? }
+          ics.each do |ic|
+            ic.data.each do |inventory_object|
+              inventory_object.data[:refresh_state_part_id] = refresh_state_part.id
+            end
+          end
+        end
+      end
+
+      # @param type [Symbol] :refresh_state_part | :refresh_state
+      def log_refresh_time_tracking(type)
+        msg = "Refresh tracking: "
+        msg += "State: #{refresh_state_uuid} "
+        msg += "| Part: #{refresh_state_part_uuid} " if type == :refresh_state_part
+
+        data = if type == :refresh_state_part
```

Comment:
If it isn't the proper part will the timestamp be nil?  Loks like we could just log all timestamps that are present

Reply:
yes, refresh_state timestamps (sweeping) are nil for refresh_state_part and vice versa . 
And "started_at" and "collected_at" from collector can be nil when it isn't merged (or new collector will come). 

"save" timestamps and "sweep" timestamps couldn't be logged together